### PR TITLE
feat: [UIE-8269], [UIE-8290] - DBaaS: Access Controls text update, fix defaultDB for legacy instances

### PR DESCRIPTION
**Pull request by mpolotsk-akamai**
## Description 📝

DBaaS: Access Controls text update, fix defaultDB for legacy instances

## Changes  🔄

- Summary Tab: updated 'Database name' for legacy instances
- Database Create > Access Controls: updated copy, placeholder, button text 
- Settings Tab > Manage Access: updated copy (only for new instances)
- Manage Access Drawer: updated copy, placeholder, button text (only for new instances)

## Target release date 🗓️

12/10/24

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-12-04 at 8 19 32 PM | Screenshot 2024-12-04 at 8 19 48 PM |
| Screenshot 2024-12-04 at 8 20 33 PM | Screenshot 2024-12-04 at 8 21 02 PM |
| Screenshot 2024-12-04 at 8 22 32 PM | Screenshot 2024-12-04 at 8 22 45 PM |
| Screenshot 2024-12-04 at 8 24 44 PM | Screenshot 2024-12-04 at 8 24 37 PM |

### Prerequisites

(How to setup test environment)
- dbaasV2 feature flag set to GA

### Verification steps

(How to verify changes)

- Select the legacy database and navigate to the Summary tab. Verify that the Database Name value is updated.
- Navigate to the Database Create > Access Controls. The copy text, placeholder, and button text are updated.
- Navigate to the Settings tab and select Manage Access. The copy text is updated (applicable only for new instances).
- On the Landing Page, click the Action Menu and choose Manage Access Controls. Verify that the Manage Access Drawer has updated copy text, placeholder, and button text (applicable only for new instances).

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

### CODE REVIEW

File: packages/manager/src/features/Databases/DatabaseDetail/AddAccessControlDrawer.tsx [hunk 1]
```diff
@@ -180,20 +188,29 @@ const AddAccessControlDrawer = (props: CombinedProps) => {
             ))
           : null}
         <Typography className={classes.instructions} variant="body1">
-          Add, edit, or remove IPv4 addresses and ranges that should be
-          authorized to access your cluster.
+          {isDefaultDB
+            ? 'Add, edit, or remove IPv6 (recommended) or IPv4 addresses or ranges that should be authorized to access your cluster.'
+            : 'Add, edit, or remove IPv4 addresses and ranges that should be authorized to access your cluster.'}{' '}
+          <Link to={learnMoreLink}>Learn more</Link>
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/AddAccessControlDrawer.tsx:194]:
```suggestion
          <Link to={learnMoreLink}>Learn more</Link>.
```

**Review by cpathipa — APPROVED** [review]
LGTM! Confirming on copy updates in Summary tab and Manage Access Controls.

File: packages/manager/src/features/Databases/DatabaseDetail/AddAccessControlDrawer.tsx [hunk 2]
```diff
@@ -166,8 +171,11 @@ const AddAccessControlDrawer = (props: CombinedProps) => {
     }
   }, [open, resetForm]);
 
+  const learnMoreLink = isDefaultDB
+    ? 'https://techdocs.akamai.com/cloud-computing/docs/aiven-manage-database#ipv6-support'
+    : 'https://techdocs.akamai.com/cloud-computing/docs/manage-access-controls';
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/AddAccessControlDrawer.tsx:176]: Ideally we'd have these URLs in a `Databases/constants.ts` file (see `VPCs/constants.ts` as an example on how to structure the file and variable names)

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettings.tsx [hunk 3]
```diff
@@ -32,8 +32,9 @@ export const DatabaseSettings: React.FC<Props> = (props) => {
 
   const accessControlCopy = (
     <Typography>
-      Add or remove IPv4 addresses or ranges that should be authorized to access
-      your cluster.
+      {isDefaultDB
+        ? 'Add or remove IPv6 (recommended) or IPv4 addresses or ranges that should be authorized to access your cluster.'
+        : 'Add or remove IPv4 addresses or ranges that should be authorized to access your cluster.'}
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettings.tsx:37]: would be good to get this copy in a `Databases/constants.ts` file also

Comment (mpolotsk-akamai) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseSettings/DatabaseSettings.tsx]: I've moved all that links and text to the new Databases/constants.ts
Thank you for the review!

**Review by dwiley-akamai — APPROVED** [review]

**Review by jdamore-linode — CHANGES_REQUESTED** [review]
@mpolotsk-akamai It looks like these changes may have broken the DBaaS update tests. Mind taking care of that before we merge? cc @cpathipa

**Comment by cpathipa** [comment]
Yes, Just verified failed tests are passed in develop, But failed on this PR @mpolotsk-akamai Could you please take a look.

**Pull request merged**